server/server.go

```diff
@@ -112,9 +117,26 @@ func (s *RpcEndPointServer) Start() {
 
 	<-notifier
 
+	s.SetUnhealthy()
+	s.logger.Info("http server stopping")
+
+	// Wait for 60 seconds for load balancer to pick up the unhealthy state or for SIGTERM to be sent again
+	select {
+	case <-time.After(60 * time.Second):
```

Comment:
**issue:**

now, we will have to wait for 60s every time we shut it down, even if it's run locally.

Reply:
agree, fixed

Reply:
unless AWS LB target group "talks" directly to the EC2's port.

but then, that would render the implementation AWS-specific.  e.g. I am not sure how exactly GCP with its VPC-native clusters handles the ingress.

Reply:
o.t.o.h., using pre-stop hook is the way AWS suggests to do it themselves:
https://aws.amazon.com/blogs/containers/how-to-rapidly-scale-your-application-with-alb-on-eks-without-losing-traffic/

Reply:
From the example in this link I get a different conclusion.

Are you sure that "TERMINATING" state is relevant here? 

What I see there is that they use pre stop hook to trigger health unavailable response. But this health unavailable will lead to
>its corresponding k8s service stops routing any more requests to it

independent on if node is terminating or not because they use readiness probe.

Do you want to test it? We have both of prs ready so it will not take a long time.
We can do it together. I think it would be interesting. I would learn something